core/Config.php

```diff
@@ -410,13 +410,15 @@ protected function writeConfig()
                 throw $this->getConfigNotWritableException();
             }
 
+            if (!$this->sanityCheck($localPath, $output)) {
+                // If sanity check fails, try to write the contents once more before logging the issue.
+                if (@file_put_contents($localPath, $output, LOCK_EX === false)) {
+                    StaticContainer::get(LoggerInterface::class)->info("The configuration file {$localPath} did not write correctly.");
```

Comment:
I think we will want to post the event here before the log call in case we adjust the above `if` statement to also check for the `sanityCheck`. Otherwise we would post the event twice. I suppose putting the event here would make it harder to test if the event was triggered and it be no longer possible to properly test it I suppose.

Reply:
@tsteur yes, I was thinking about sanity check in this above `if` too and moving the event to be posted within, but it was tricky for the tests. I have added an optional 3rd argument `$notify` to the `sanityCheck()` function which allows us to set this to true on the second try, and this will trigger the event to be posted. This should prevent the event being posted twice, and also keep testability.